### ISSUE TITLE
Greedy sampling?

### ISSUE THREAD

**Issue by AmgadHasan** [theroyallab/tabbyAPI#152]
Hi

Is there a way to do greedy sampling using the completion and chat completion endpoints?

ExLLAMAv2 supports greedy sampling using `ExLlamaV2Sampler.Settings.greedy()`

**Comment by DocShotgun** [comment]
`top_k` = 1

**Comment by AmgadHasan** [comment]
> `top_k` = 1
I just tested this out and it generates a different response each time

**Comment by DocShotgun** [comment]
> > `top_k` = 1
> > I just tested this out and it generates a different response each time

https://github.com/turboderp/exllamav2/issues/232#issuecomment-1860896496

**Comment by AmgadHasan** [comment]
> > > `top_k` = 1
> > > I just tested this out and it generates a different response each time
> 
> [turboderp/exllamav2#232 (comment)](https://github.com/turboderp/exllamav2/issues/232#issuecomment-1860896496)
> 
> tl;dr: the launch order of CUDA threads is non-deterministic, and the intermediate results are rounded to fp16 precision, so sometimes enough rounding error occurs to flip which token is the top probability token, and thus even with "greedy" sampling, the result isn't guaranteed to be the same on each generation

Thanks! This explains it
